org.rcpml.swt/src/org/rcpml/swt/SWTComboBridge.java

```diff
@@ -97,15 +158,25 @@ public void update() {
 			}
 			if (!gfound) {
 				this.fCombo.setItems(items3);
-				this.fCombo.select(selection);
+			} 
+			if (selection/2 != fCombo.getSelectionIndex() && !state.equals(""))
```

Comment:
Please remove this. selection/2 will not always work for particular inputs.
To calculate selection, create and remember a counter, which counting Element nodes.